### ISSUE TITLE
Support for enhanced notification format

### ISSUE THREAD

**Issue by GoogleCodeExporter** [ntrees/javapns#46]
```
I do not see the support for enhanced format from the source code. Did I miss 
something? 

I guess I can modify getMessage() for enhanced format, but not sure how to 
handle No-Error or any timeout needed to retrieve the response.

Any sample code will be appreciated.

Weiping
```

Original issue reported on code.google.com by `wguo1...@gmail.com` on 12 Apr 2011 at 9:05

**Comment by GoogleCodeExporter** [comment]
```
I am also interested in this.  Specifically the expiry information.  We want 
our Notifications to expire at a certain time after they are initially created. 

http://developer.apple.com/library/ios/#documentation/NetworkingInternet/Concept
ual/RemoteNotificationsPG/CommunicatingWIthAPS/CommunicatingWIthAPS.html%23//app
le_ref/doc/uid/TP40008194-CH101-SW4
```

Original comment by `akin...@gmail.com` on 27 Apr 2011 at 8:31

**Comment by GoogleCodeExporter** [comment]
```
Note:  current library does use the enhanced notification format, but the 
identifier and expiry values are hardcoded.  Future enhancements could allow 
for those two values to be customized.
```

Original comment by `sype...@gmail.com` on 9 Sep 2011 at 7:17
- Changed title: **Support for enhanced notification format**
- Added labels: **Type-Enhancement**
- Removed labels: **Type-Defect**

**Comment by GoogleCodeExporter** [comment]
```
Fixed in 2.0 Beta 4.   See new Payload.setExpiry(..) method, and the 
introduction of a new identifier parameter in a few 
PushNotificationManager.sendNotification(..) methods.  Automatic identifier 
generation is also supported in the javapns.notification.transmission classes.

See issue #47 for enhancement request related to error packets.
```

Original comment by `sype...@gmail.com` on 15 Sep 2011 at 7:24
- Changed state: **Fixed**
